### DISCUSSION TITLE
DietPi-Software | Navidrome: Lightweight alternative to Airsonic

### DISCUSSION THREAD

**Discussion by bbsixzz** [MichaIng/DietPi#5095] [Software Request] [answered]
https://www.navidrome.org/

Up-and-coming airsonic killer, using less resources; modern GoLang instead of Java.

Supports subsonic API!

Vote for it with the up button at the post.

**Comment by MichaIng** [comment]
Many thanks for your request.

Looks reasonable, as the massive startup delay and memory consumption of the Java runtime is really the reason I'd personally currently never use Subsonic nor Airsonic.

**Comment by bbsixzz** [comment]
Great, I'm holding off for the same reason. I'm using the free iSub app on iOS but have heard great things about play:sub (not free).

**Comment by bbsixzz** [comment]
Still only one vote on feathub :/

Come on everyone with airsonic should want this!

**Comment by bbsixzz** [comment]
I just deployed this and it works excellent following the official install guide https://www.navidrome.org/docs/installation/linux/

I configured everything like this to fit DietPi:

```
++++++ Install prerequisites

sudo apt install ffmpeg

++++++ Create directory structure

sudo install -d -o dietpi -g dietpi /opt/navidrome
sudo install -d -o dietpi -g dietpi /var/lib/navidrome

++++++ Get Navidrome

wget https://github.com/navidrome/navidrome/releases/download/v0.44.1/navidrome_0.44.1_Linux_armv6.tar.gz -O Navidrome.tar.gz
sudo tar -xvzf Navidrome.tar.gz -C /opt/navidrome/
sudo chown -R dietpi:dietpi /opt/navidrome

++++++ Create configuration file navidrome.toml in the working directory /var/lib/navidrome

DataFolder = '/var/lib/navidrome/data'
MusicFolder = '/mnt/dietpi_userdata/Music'
ImageCacheSize = '100MiB'

LogLevel = 'info'
SessionTimeout = '24h'
AuthRequestLimit = '5'
AuthWindowLength = '20s'
ScanSchedule = '@every 1h'
EnableLogRedacting = 'true'
Scanner.Extractor = 'taglib'

CoverJpegQuality = '75'
CoverArtPriority = 'embedded'

++++++ Create a systemd unit under /etc/systemd/system/ named navidrome.service

[Unit]
Description=Navidrome Music Server and Streamer compatible with Subsonic/Airsonic
After=remote-fs.target network.target
AssertPathExists=/var/lib/navidrome

[Install]
WantedBy=multi-user.target

[Service]
User=dietpi
Group=dietpi
Type=simple
ExecStart=/opt/navidrome/navidrome --configfile "/var/lib/navidrome/navidrome.toml"
WorkingDirectory=/var/lib/navidrome
TimeoutStopSec=20
KillMode=process
Restart=on-failure

# See https://www.freedesktop.org/software/systemd/man/systemd.exec.html
DevicePolicy=closed
NoNewPrivileges=yes
PrivateTmp=yes
PrivateUsers=yes
ProtectControlGroups=yes
ProtectKernelModules=yes
ProtectKernelTunables=yes
RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
RestrictNamespaces=yes
RestrictRealtime=yes
SystemCallFilter=~@clock @debug @module @mount @obsolete @reboot @setuid @swap
ReadWritePaths=/var/lib/navidrome

# You can uncomment the following line if you're not using the jukebox This
# will prevent navidrome from accessing any real (physical) devices
#PrivateDevices=yes

# You can change the following line to `strict` instead of `full` if you don't
# want navidrome to be able to write anything on your filesystem outside of
# /var/lib/navidrome.
ProtectSystem=full

# You can uncomment the following line if you don't have any media in /home/*.
# This will prevent navidrome from ever reading/writing anything there.
#ProtectHome=true

# You can customize some Navidrome config options by setting environment variables here. Ex:
#Environment=ND_BASEURL="/navidrome"

++++++ Start the Navidrome Service

sudo systemctl daemon-reload
sudo systemctl start navidrome.service
sudo systemctl status navidrome.service

++++++ Start Navidrome on Startup

sudo systemctl enable navidrome.service
```

Works great without ffmpeg right out of the box, but you'll need it if you want transcoding.

Here's hoping for inclusion in DietPi 7.5 :]

**Comment by bbsixzz** [comment]
I updated Navidrome to v0.45.0 using the following procedure:

```
sudo systemctl stop navidrome.service

wget https://github.com/navidrome/navidrome/releases/download/v0.45.0/navidrome_0.45.0_Linux_armv6.tar.gz  -O Navidrome.tar.gz

sudo tar -xvzf Navidrome.tar.gz -C /opt/navidrome/

sudo systemctl start navidrome.service
```

Everything is working fine, is there any way to streamline this?

**Comment by jake-g** [comment]
was looking into this as a non-java alt for airsonic (which ive been using for years), then found this thread. I'll try to install manually, but would be great to see this officially supported (assuming it works well for me)

and thanks for the instructions  @bbsixzz

**Comment by MichaIng** [comment]
I'll move this over to discussions, so you guys can vote for it.

**Comment by LeatherMolly** [comment]
Hello MichIng and all the participators of that discussion/ voting.

I used some time ago DietPi with Pihole - and thanks to the dev (s), what a wonderful piece of development ;-)

Actually I want to use navidrome and found out that there are some who would like to see it be integrated into DietPi.

I support this idea, because I am a user coming out of the non-tecnical world/ non-IT world and for me it helps a lot, when good people use their programming knowledge to deploy s. th. what I can use without reading days & nights over weeks ans weeks to get it to work.
Beein around in the linux desktop world since over 10 years, knowing some, reading some, but f.e. not firm with docker & co....

So - Thanks if it will find iits way into a "quick installation" section into the DietPi system. Daumen hoch!!!

Kind regards
O.

**Comment by pedrom34** [comment]
Great suggestion. I hope it's going to be supported as Airsonic-advanced use so much memory and isn't updated anymore...

**Comment by Joulinar** [comment, marked as answer]
Will be part of DietPi v8.7 https://github.com/MichaIng/DietPi/pull/5627
